### ISSUE TITLE
reset of navigation tools when we switch views

### ISSUE THREAD

**Issue by touretzkyds** [touretzkyds/KnowledgeGraphDemo#56]
Type "Iguana" in the search box and click on the taxon node to get into the property view. Now in the search box type a totally unrelated taxon name, such as "Lion". The view switches back to hierarchy view and shows Lion and its parents, but the navigation history and navigation arrows still show data for Iguana. They should be reset when we switch back to the hierarchy view.

We don't actually have to hide the navigation view. It can be useful to display it. But it should show the newly-selected node and its parents, e.g., when we type "Lion" in the search box it should show us the path from Lion to the root. This saves us the trouble of having to zoom in far enough to make all the taxon names visible.

Similarly, we could leave the navigation arrows active in hierarchy view. Clicking on an arrow simply changes the selected node (the one shown in red) and updates the navigation history.

**Comment by nhadlaw** [comment]
The latest version of demo2d.html should have fixed the bug. In regards to the second issue (enhancement) what I did was I added the feature so that if a user is just viewing the hierarchy, the nav button will perform the query based on if they move up down, left or right on the respective node. Rather, in the property display mode, if a user moves up/down/left/right, the properties of that respective node are shown. I did this because when we first see the nav button after we enter a taxon in the textbox, it doesn't display it's properties but rather shows it as a red node in a larger graph. If a user moves up/down/left/right, if we want to mimic what we saw when the first taxon was entered, I think the best way to do that is to replicate what we did for that taxon. What I did should make more sense when you test out the demo but if you would like me to change it to always show properties when using the nav buttons then that's an easy fix.

**Comment by touretzkyds** [comment]
This is good! But it's only allowing up/down navigation. Can we enable sibling navigation as well?

**Comment by nhadlaw** [comment]
The reason there is no right/left navigation is that when a user enters a taxon into the box, we show everything above and below it so we don't have any siblings of that taxon shown. Thus, the nav button doesn't have right/left navigation of the current taxon shown. We could change this to have the user able to go to it's siblings but I just didn't know if we wanted to do that because the hierarchy that the user sees, after a specific taxon is entered, doesn't display the siblings.

**Comment by touretzkyds** [comment]
Good point. So my suggestion is to offer the siblings in the navigation arrow, and if the use goes to a sibling, we regenerate the tree the same as if they had typed that node name into the search box.
